api_app/analyzers_manager/observable_analyzers/virushee.py

```diff
@@ -0,0 +1,40 @@
+# This file is a part of IntelOwl https://github.com/intelowlproject/IntelOwl
+# See the file 'LICENSE' for copying permission.
+
+import requests
+
+from api_app.analyzers_manager.classes import ObservableAnalyzer
+from api_app.exceptions import AnalyzerRunException
+from tests.mock_utils import MockResponse, if_mock_connections, patch
+
+
+class VirusheeCheckHash(ObservableAnalyzer):
+    base_url: str = "https://api.virushee.com/file/hash/{input}"
+
+    def run(self):
+        url = self.base_url.format(input=self.observable_name)
+        api_key = self._secrets["api_key_name"]
+        if api_key:
+            headers = {"X-API-Key": api_key}
+        else:
+            headers = {}
```

Comment:
this should be part of the `set_params` method. Create a `requests.Session` object passing the headers.